src/main/java/com/mindee/MindeeClient.java

```diff
@@ -490,6 +491,280 @@ private PredictResponse<CustomV1> parse(
+      Class<T> type,
+      Endpoint endpoint,
+      URL sourceUrl
+  ) throws IOException {
+    InputSourceUtils.validateUrl(sourceUrl);
+    return this.enqueue(
+      type,
+      endpoint,
+      null,
+      null,
+      null,
+      sourceUrl
+    );
+  }
+
+  /**
+   * Send a remote file to a Generated prediction async API queue.
+   */
+  public <T extends GeneratedV1> AsyncPredictResponse<T> enqueue(
+      Class<T> type,
+      Endpoint endpoint,
+      URL sourceUrl,
+      PredictOptions predictOptions
+  ) throws IOException {
+    InputSourceUtils.validateUrl(sourceUrl);
+    return this.enqueue(
+      type,
+      endpoint,
+      null,
+      null,
+      predictOptions,
+      sourceUrl
+    );
+  }
+
+
+  /**
+   * Send a local file to a Generated prediction API async queue, poll, and parse when complete.
+   */
+  public <T extends GeneratedV1> AsyncPredictResponse<T> enqueueAndParse(
+      Class<T> type,
+      Endpoint endpoint,
+      LocalInputSource localInputSource
+  ) throws IOException, InterruptedException {
+    return this.enqueueAndParse(
+      type,
+      endpoint,
+      null,
+      localInputSource.getFile(),
+      localInputSource.getFilename(),
+      null,
+      null);
+  }
+
+  /**
+   * Send a local file to a Generated prediction API async queue, poll, and parse when complete.
+   */
+  public <T extends GeneratedV1> AsyncPredictResponse<T> enqueueAndParse(
+      Class<T> type,
+      Endpoint endpoint,
+      LocalInputSource localInputSource,
+      AsyncPollingOptions pollingOptions
+  ) throws IOException, InterruptedException {
+    return this.enqueueAndParse(
+      type,
+      endpoint,
+      pollingOptions,
+      localInputSource.getFile(),
+      localInputSource.getFilename(),
+      null,
+      null
+    );
+  }
+
+  /**
+   * Send a local file to a Generated prediction API async queue, poll, and parse when complete.
+   */
+  public <T extends GeneratedV1> AsyncPredictResponse<T> enqueueAndParse(
+      Class<T> type,
+      Endpoint endpoint,
+      LocalInputSource localInputSource,
+      PredictOptions predictOptions,
+      PageOptions pageOptions,
+      AsyncPollingOptions pollingOptions
+  ) throws IOException, InterruptedException {
+    return this.enqueueAndParse(
+      type,
+      endpoint,
+      pollingOptions,
+      getSplitFile(localInputSource, pageOptions),
+      localInputSource.getFilename(),
+      predictOptions,
+      null
+    );
+  }
+
+  /**
+   * Send a remote file to a Generated prediction API async queue, poll, and parse when complete.
+   */
+  public <T extends GeneratedV1> AsyncPredictResponse<T> enqueueAndParse(
+      Class<T> type,
+      Endpoint endpoint,
+      URL sourceUrl
+  ) throws IOException, InterruptedException {
+    InputSourceUtils.validateUrl(sourceUrl);
+    return this.enqueueAndParse(
+      type,
+      endpoint,
+      null,
+      null,
+      null,
+      null,
```

Comment:
Yeah...

Reply:
the private function is private for a reason ;-)